static/js/ui_init.js

```diff
@@ -280,7 +280,14 @@ $(function () {
     compose.initialize();
     hotspots.initialize();
     ui.initialize();
-    desktop_notifications_panel.initialize();
+
+    // if email has not been set up and the user is the admin, display a warning
+    // to tell them to set up an email server.
+    if (page_params.warn_no_email === true && page_params.is_admin) {
+        panels.initialize($("[data-process='email-server']"));
+    } else {
+        panels.initialize($("[data-process='notifications']"));
+    }
```

Comment:
Can we push this logic inside `panels.js`? E.g., rename the existing `initialize` to something like `initialize_process`, and make a small new `initialize` that just has these lines. Or maybe stick this conditional at the top of `initialize` and have it set a local like `$process`.

That way all of this stuff is inside the same file and close together, which feels preferable since it's all pretty closely tied together. Also `ui_init.js` gets to continue not caring about these pretty specific details.